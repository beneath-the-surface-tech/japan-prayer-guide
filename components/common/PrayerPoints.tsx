import React from "react"
import { Container, Card } from "react-bootstrap"
import { FaPrayingHands } from "react-icons/fa"
import { IconContext } from "react-icons/lib"
import { BsDownload } from "react-icons/bs"
import Button from "react-bootstrap/Button"
import { useTranslation } from "next-i18next"

interface prayerProps {
    prayerPoints: string[] // array of prayer points
    title?: string // Title that goes on top of card
    featured?: boolean // is this component the "featured" component?
    showImg?: boolean // show the image?
    showSubtitle?: boolean // show the "Pray For" subtitle?
}

export default function PrayerPoints({
    prayerPoints,
    title,
    featured = false,
    showImg = false,
    showSubtitle = false,
}: prayerProps) {
    const { t } = useTranslation("common")

    const subtitle: string = t("prayerSummary.subtitle")
    const view: string = t("prayerSummary.viewAll")
    const read: string = t("prayerSummary.readMore")

    const displayTitle: string = title ? title : t("prayerSummary.title")

    return (
        <Container
            data-testid={"prayer-points-container"}
            className={"d-flex px-6" + (showSubtitle ? " justify-content-center" : " justify-content-start")}
            id="topic-prayer"
        >
            <Card className="my-4 shadow d-md-flex flex-md-row">
                {/* temporary height for video, before we finalize videos */}
                {showImg && (
                    <Card.Img
                        className="h-100"
                        variant="top"
                        src="/bamboo.jpeg"
                        style={{ maxWidth: "600px" }}
                    ></Card.Img>
                )}
                <Card.Body style={showSubtitle ? { maxWidth: "600px" } : { maxWidth: "400px" }}>
                    <Card.Text
                        data-testid={"prayer-points-title"}
                        className="px-2 pb-3 fs-2 fst-italic fw-bold border-bottom border-grey d-flex justify-content-between align-items-center"
                    >
                        {displayTitle}
                        <IconContext.Provider value={{ size: "30px" }}>
                            <BsDownload className="text-secondary fw-bold" style={{ cursor: "pointer" }}></BsDownload>
                        </IconContext.Provider>
                    </Card.Text>
                    {showSubtitle && (
                        <Card.Text className="d-flex align-items-center gap-2 text-secondary">
                            <IconContext.Provider value={{ size: "20px" }}>
                                <FaPrayingHands></FaPrayingHands>
                            </IconContext.Provider>
                            {subtitle}
                        </Card.Text>
                    )}
                    <ul className="fs-5" data-testid={"prayer-points-points"}>
                        {prayerPoints.map((point: string, idx: number) => (
                            <li key={idx + point} className="my-3">
                                {point}
                            </li>
                        ))}
                    </ul>
                    {featured && (
                        <>
                            <Button className="w-100 mt-2 text-secondary border-secondary bg-white" variant="primary">
                                {read}
                            </Button>
                            <Card.Text
                                className="my-3 w-100 mx-auto text-center text-decoration-underline text-secondary"
                                style={{ cursor: "pointer" }}
                            >
                                {view}
                            </Card.Text>
                        </>
                    )}
                </Card.Body>
            </Card>
        </Container>
    )
}
