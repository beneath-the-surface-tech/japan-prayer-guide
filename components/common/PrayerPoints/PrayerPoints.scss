// -------------------------
// Prayer Points
// -------------------------

#topic-prayer {
    $featured-card-width: 500px;
    $featured-card-width-tablet: 375px;
    $featured-card-width-mobile: 600px;

    .card-body {
        &.featured {
            max-width: $featured-card-width;
            @include media-breakpoint-down(lg) {
                max-width: $featured-card-width-tablet;
            }
            @include media-breakpoint-down(md) {
                max-width: $featured-card-width-mobile;
            }
        }

        &.topicTop {
            max-width: 452px;
            @include media-breakpoint-down(lg) {
                max-width: fit-content;
            }
        }
    }

    .prayer-img {
        height: auto;
        max-width: $featured-card-width;
        object-fit: cover;
        @include media-breakpoint-down(lg) {
            max-width: $featured-card-width-tablet;
        }
        @include media-breakpoint-down(md) {
            max-width: $featured-card-width-mobile;
        }
    }

    .prayer-title {
        font-family: $font-family-secondary;
        font-weight: 700;
    }

    .prayerSubtitle {
        font-family: $font-family-secondary;
        font-weight: 700;
        font-size: 16px;
        @include media-breakpoint-down(md) {
            font-size: 14px;
        }
    }
}

.prayer-points-card {
    border-radius: 4px;
    @include media-breakpoint-up(lg) {
        max-width: 646px;
    }
}

.prayerSubtitle {
    font-family: $font-family-secondary;
    font-weight: 700;
    font-size: 20px;
}

.prayer-title {
    font-family: $font-family-secondary;
    font-weight: 700;
    font-size: 24px;
}

#topic-prayer-featured {
    .card-body {
        background-color: white;
        @include media-breakpoint-up(xl) {
            max-width: 550px;
        }
    }

    .prayer-img {
        object-fit: cover;
        @include media-breakpoint-up(md) {
            width: 50%;
        }
        @include media-breakpoint-up(xl) {
            height: 100%;
        }
        @include media-breakpoint-up(xl) {
            width: calc(100% - 550px);
        }
    }

    .prayer-points-card {
        @include media-breakpoint-up(lg) {
            max-width: none;
        }
    }
}
