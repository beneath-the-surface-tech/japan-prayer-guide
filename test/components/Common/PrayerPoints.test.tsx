import { render, screen } from "@testing-library/react"
import PrayerPoints from "../../../components/common/PrayerPoints"

describe("Prayer Points", () => {
    test("Snapshot test", () => {
        // This test is to check against unintended changes.
        // If the change is intentional you can update the snapshot with `jest --updateSnapshot`

        const testTextArray = ["ABC", "Item123"]
        const testTitle = "Prayer Points1"
        const component = render(<PrayerPoints prayerPoints={testTextArray} title={testTitle} />)
        expect(component).toMatchSnapshot()
    })

    test("Renders a section with the right text inside", () => {
        const testTextArray = ["Hey how's it going", "I'm doing fine", "Thanks"]
        const testTitle = "Prayer Points"
        render(<PrayerPoints prayerPoints={testTextArray} title={testTitle} />)
        const TopicPrayerPointsCont = screen.getByTestId("prayer-points-container")
        const TopicPrayerPointsTitle = screen.getByTestId("prayer-points-title")
        const TopicPrayerPointsPoints = screen.getByTestId("prayer-points-points").children

        expect(TopicPrayerPointsCont).toHaveClass("d-flex", "container", "px-6")

        expect(TopicPrayerPointsTitle).toHaveTextContent(testTitle)
        expect(TopicPrayerPointsTitle).toHaveClass("px-2", "pb-3", "d-flex")

        expect(TopicPrayerPointsPoints.length).toBe(3)
        expect(TopicPrayerPointsPoints[0]).toHaveTextContent(testTextArray[0])
        expect(TopicPrayerPointsPoints[0]).toHaveClass("my-3")
        expect(TopicPrayerPointsPoints[1]).toHaveTextContent(testTextArray[1])
        expect(TopicPrayerPointsPoints[1]).toHaveClass("my-3")
        expect(TopicPrayerPointsPoints[2]).toHaveTextContent(testTextArray[2])
        expect(TopicPrayerPointsPoints[2]).toHaveClass("my-3")
    })
})
