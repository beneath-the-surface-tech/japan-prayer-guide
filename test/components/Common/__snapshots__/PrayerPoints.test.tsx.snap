// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prayer Points Snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex px-6 justify-content-center container"
        data-testid="prayer-points-container"
        id="topic-prayer-featured"
      >
        <div
          class="shadow d-md-flex flex-md-row my-4 prayer-points-card card"
        >
          <img
            class="card-img-top prayer-img"
            src="/photos/topic-nav/church/church-leadership-b.jpg"
          />
          <div
            class="p-4 featured card-body"
            data-testid="prayer-points-body"
          >
            <p
              class="px-2 pb-2 fs-2 border-bottom border-grey prayer-title card-text"
              data-testid="prayer-points-title"
            >
              title
            </p>
            <p
              class="fs-3 prayerSubtitle d-flex align-items-center gap-2 text-secondary-5 card-text"
            >
              <img
                alt="praying hands"
                class=""
                src="/icons/prayer.png"
                style="height: 20px;"
              />
              prayerSummary.subtitle
            </p>
            <ul
              class="bullet-points"
              data-testid="prayer-points-points"
            >
              <li
                class="my-2 bullet-point"
              >
                A
              </li>
              <li
                class="my-2 bullet-point"
              >
                B
              </li>
            </ul>
            <button
              class="feature-button fs-4 fw-bold bg-grey-4 text-white text-center border-0 mt-3 mb-2 w-100 btn btn-primary"
              type="button"
            >
              prayerSummary.readMore
            </button>
            <p
              class="mt-3 w-100 mx-auto text-center text-decoration-underline text-secondary-5 fw-bold fs-4 feature-view-all card-text"
              style="cursor: pointer;"
              tabindex="0"
            >
              prayerSummary.viewAll
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex px-6 justify-content-center container"
      data-testid="prayer-points-container"
      id="topic-prayer-featured"
    >
      <div
        class="shadow d-md-flex flex-md-row my-4 prayer-points-card card"
      >
        <img
          class="card-img-top prayer-img"
          src="/photos/topic-nav/church/church-leadership-b.jpg"
        />
        <div
          class="p-4 featured card-body"
          data-testid="prayer-points-body"
        >
          <p
            class="px-2 pb-2 fs-2 border-bottom border-grey prayer-title card-text"
            data-testid="prayer-points-title"
          >
            title
          </p>
          <p
            class="fs-3 prayerSubtitle d-flex align-items-center gap-2 text-secondary-5 card-text"
          >
            <img
              alt="praying hands"
              class=""
              src="/icons/prayer.png"
              style="height: 20px;"
            />
            prayerSummary.subtitle
          </p>
          <ul
            class="bullet-points"
            data-testid="prayer-points-points"
          >
            <li
              class="my-2 bullet-point"
            >
              A
            </li>
            <li
              class="my-2 bullet-point"
            >
              B
            </li>
          </ul>
          <button
            class="feature-button fs-4 fw-bold bg-grey-4 text-white text-center border-0 mt-3 mb-2 w-100 btn btn-primary"
            type="button"
          >
            prayerSummary.readMore
          </button>
          <p
            class="mt-3 w-100 mx-auto text-center text-decoration-underline text-secondary-5 fw-bold fs-4 feature-view-all card-text"
            style="cursor: pointer;"
            tabindex="0"
          >
            prayerSummary.viewAll
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
